orocos_kdl/src/CMakeLists.txt

```diff
@@ -126,14 +126,15 @@ ENDIF()
 #####end RPATH
 
 # Needed so that the generated config.h can be used
-INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
-TARGET_LINK_LIBRARIES(orocos-kdl ${Boost_LIBRARIES})
+TARGET_INCLUDE_DIRECTORIES(orocos-kdl PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}> )
+TARGET_LINK_LIBRARIES(orocos-kdl PUBLIC ${Boost_LIBRARIES} Eigen3::Eigen)
```

Comment:
This target was introduced in Eigen 3.3.1, you'll need to add some fallback behavior (`if(TARGET Eigen3::Eigen)...else()...endif()`).

Reply:
Sorry for the (very) long absence, is this still relevant?

Reply:
Since KDL doesn't enforce a minimum version, anyone trying to use Eigen<3.3.1 will experience errors. However, you could just tweak [FindEigen3.cmake](https://github.com/orocos/orocos_kinematics_dynamics/blob/73721147ef7339df11a6a587f15db73410168873/orocos_kdl/cmake/FindEigen3.cmake) so that modern CMake targets are made available for use both by KDL itself and projects consuming KDL (make sure [this](https://github.com/orocos/orocos_kinematics_dynamics/blob/73721147ef7339df11a6a587f15db73410168873/orocos_kdl/orocos_kdl-config.cmake.in#L10-L14) gives precedence to the local file or just add `OR NOT TARGET Eigen3::Eigen` in the `if` clause). See [upstream find module](https://gitlab.com/libeigen/eigen/-/blob/db61b8d47825e55e348a66e77cb9e1f1cbae066b/cmake/FindEigen3.cmake), especially the last few lines:

```cmake
if(EIGEN3_FOUND AND NOT TARGET Eigen3::Eigen)
  add_library(Eigen3::Eigen INTERFACE IMPORTED)
  set_target_properties(Eigen3::Eigen PROPERTIES
    INTERFACE_INCLUDE_DIRECTORIES "${EIGEN3_INCLUDE_DIR}")
endif()
```

Reply:
currently if Eigen is not found there is no error? Adding eigen with the `TARGET_LINK_LIBRARIES` will print an error when generating.

Reply:
If Eigen is a required dependency, there should be a check somewhere else (and *before* those lines) to issue an error. I wouldn't delegate such responsibility to the `target_link_libraries` call, typically you would use `find_package(<pkg> REQUIRED)` or a simple `if(<pkg>_FOUND)`-then-`message(FATAL_ERROR ...)` condition instead.